### PR TITLE
Better document parallel::TriangulationBase.

### PR DESCRIPTION
**Pull request by bangerth**

### CODE REVIEW

File: include/deal.II/distributed/tria_base.h [hunk 1]
```diff
@@ -40,8 +40,37 @@ namespace parallel
...
+   * It is, consequently, a class that can be used to test whether a
+   * pointer of reference to a triangulation object refers to a
+   * sequential triangulation, or whether the triangulation is in fact
+   * parallel. In other words, one could write a function like this:
+   * @code
+   *   template <int dim, int spacedim>
+   *   bool is_parallel (const dealii::Triangulation<dim,spacedim> &tria)
+   *   {
+   *     if (dynamic_cast<const
+   * parallel::TriangulationBase<dim,spacedim>*>(&tria)
```
Comment (kronbichler) [inline include/deal.II/distributed/tria_base.h]: Would you mind adding some indentation in the comment here? (I believe the indentation script inserted the break here.)

File: include/deal.II/distributed/tria_base.h [hunk 2]
```diff
@@ -40,8 +40,37 @@ namespace parallel
...
+   *         != nullptr)
+   *       return true;
+   *     else
+   *       return false;
+   *   }
+   * @endcode
+   *
+   * All parallel triangulations share certain traits, such as the fact that
+   * they communicate via @ref GlossMPICommunicator "MPI communicators",
```
Comment (kronbichler) [inline include/deal.II/distributed/tria_base.h]:
```suggestion
   * they communicate via @ref GlossMPICommunicator "MPI communicators" or
```

**Comment by peterrum** [comment]
Thanks @bangerth !

File: include/deal.II/distributed/tria_base.h (hunk 1 above)
Comment (bangerth) [reply include/deal.II/distributed/tria_base.h]: I'm never going to learn to love this tool... :-(

**Comment by bangerth** [comment]
Changes made!

**Pull request merged**